src/rust/ide/src/controller/module.rs

```diff
@@ -25,13 +42,56 @@ use crate::double_representation::definition::DefinitionName;
 // ============
 
 /// Path identifying module's file in the Language Server.
-pub type Path = language_server::Path;
+#[derive(Clone,Debug,Eq,Hash,PartialEq)]
+pub struct Path {
+    file_path : FilePath,
+}
+
+impl Path {
+    /// Create a path from the file path. Returns None if given path is not a valid module file.
+    pub fn from_file_path(file_path:FilePath) -> Option<Self> {
+        let has_proper_ext   = file_path.extension() == Some(constants::LANGUAGE_FILE_EXTENSION);
+        let first_char       = file_path.file_name().and_then(|name| name.chars().next());
+        let capitalized_name = first_char.map_or(false, |ch| ch.is_uppercase());
+        let is_module        = has_proper_ext && capitalized_name;
+        is_module.and_option_from(|| Some(Path{file_path}))
+    }
+
+    /// Get the file path.
+    pub fn file_path(&self) -> &FilePath {
+        &self.file_path
+    }
+
+    /// Get the module name from path.
+    ///
+    /// The module name is a filename without extension.
+    pub fn module_name(&self) -> &str {
+        // The file stem existence should be checked during construction.
+        self.file_path.file_stem().unwrap()
+    }
 
-/// Gets the module name from its path.
-pub fn obtain_module_name_from_path(path:&Path) -> Option<&str> {
-    let segment     = path.segments.last()?;
-    let name_length = segment.rfind('.').unwrap_or_else(|| segment.len());
-    Some(&segment[..name_length])
+    /// Create a simple module path basing on module name. Used as test utility.
```

Comment:
I don't like "simple module path" term, as I'm not really sure what this exactly is.

Consider more like:
```
Create a module path consisting of a single segment, based on a given module name.
```